experimental/ir/lower_walk.go

```diff
@@ -47,6 +48,10 @@ func (w *walker) walk() {
 	}
 	w.pkg = w.Package().ToAbsolute()
 
+	if syn := w.AST().Syntax(); !syn.IsZero() {
+		c.syntax = syntax.Lookup(syn.Value().AsLiteral().Text())
```

Comment:
This isn't quite right. Text() returns the literal text; you need to use AsString() to get the string contents.

```go
unquoted, _ := w.AST().Syntax().Value().AsLiteral().AsString()
c.syntax = syntax.Lookup(unquoted)
```

Important to note here that this long call chain returns `""` if the syntax is missing or is something like `syntax = 42;`

Reply:
Ah yeah, this makes sense -- otherwise it would return the `token.Text()` verbatim.